routers/web/user/setting/profile.go

```diff
@@ -57,45 +55,33 @@ func HandleUsernameChange(ctx *context.Context, user *user_model.User, newName s
 		return fmt.Errorf(ctx.Tr("form.username_change_not_local_user"))
 	}
 
-	// Check if user name has been changed
-	if user.LowerName != strings.ToLower(newName) {
-		if err := user_model.ChangeUserName(user, newName); err != nil {
-			switch {
-			case user_model.IsErrUserAlreadyExist(err):
-				ctx.Flash.Error(ctx.Tr("form.username_been_taken"))
-			case user_model.IsErrEmailAlreadyUsed(err):
-				ctx.Flash.Error(ctx.Tr("form.email_been_used"))
-			case db.IsErrNameReserved(err):
-				ctx.Flash.Error(ctx.Tr("user.form.name_reserved", newName))
-			case db.IsErrNamePatternNotAllowed(err):
-				ctx.Flash.Error(ctx.Tr("user.form.name_pattern_not_allowed", newName))
-			case db.IsErrNameCharsNotAllowed(err):
-				ctx.Flash.Error(ctx.Tr("user.form.name_chars_not_allowed", newName))
-			default:
-				ctx.ServerError("ChangeUserName", err)
-			}
-			return err
-		}
-	} else {
+	if strings.EqualFold(user.LowerName, newName) {
 		if err := repo_model.UpdateRepositoryOwnerNames(user.ID, newName); err != nil {
 			ctx.ServerError("UpdateRepository", err)
 			return err
 		}
+		return nil
```

Comment:
I don't think the logic is the same as before. Previously, following steps will also be exeuted.

Reply:
you're right, turns out `user_model.ChangeUserName` already calls this, and so no need to have it duplicated.